### ISSUE TITLE
BraintreeCore Fails to Build in Xcode 16 Beta 3 with Error: Type 'UIApplication' does not conform to protocol 'URLOpener'

### ISSUE THREAD

**Issue by RGG-jayoung** [braintree/braintree_ios#1359]
### Braintree SDK Version

6.17.10 - 6.22.0

### Environment

Both

### Xcode Version

Xcode 16 Beta 3

### OS Version & Device

iOS 18

### Integration type

Swfit Package Manager

### Development Processor

Apple Silicon (M-series chips)

### Describe the bug

After updating to Apple's latest Beta 3, BraintreeCore fails to build due the following compiler error:

```
// Error Message
Type 'UIApplication' does not conform to protocol 'URLOpener'
// Fixit Suggestion
Candidate has non-matching type '(URL, [UIApplication.OpenExternalURLOptionsKey : Any], (@MainActor @Sendable (Bool) -> Void)?) -> Void' (UIKit.UIApplication)
```

It seems that the `UIApplication+URLOpener.swift` conformance to URLOpener requires stubs to fulfill it's contract.

### To reproduce

1. Open a project configured with the Braintree SDK with Xcode 16 Beta 3
2. Attempt to build an App Target that imports BraintreeCore.
3. Build will fail with the error.

Other steps taken:

- Attempted build with Braintree 6.17.0, same result.
- Attempted build with Braintree 6.22.0, same result but more protocol conformance stubs listed.

### Expected behavior

App target should build without issue like Xcode 16 Beta 1 and Xcode 16 Beta 2

### Screenshots

<img width="536" alt="Braintree Core - Application URLOpener Conformance" src="https://github.com/braintree/braintree_ios/assets/53495188/41d86830-3e34-44c0-99ad-e01de90b3bcf">

<img width="1179" alt="Braintree Core - Application URLOpener Confirmance - SDK 6_17_0" src="https://github.com/braintree/braintree_ios/assets/53495188/487cc691-291b-4a0c-9958-23ffeeb128e0">

<img width="1210" alt="Braintree Core - Application File" src="https://github.com/braintree/braintree_ios/assets/53495188/f0bfeaf5-1b94-40fe-80ca-301e0358657d">

**Comment by jaxdesmarais** [comment]
Hey @RGG-jayoung -

Thanks for bringing this issue to our attention! The method signature for this changed in Xcode 16 beta 3 ([apple dev docs diff here](https://developer.apple.com/documentation/uikit/uiapplication/1648685-open?changes=latest_minor)). Since changing to the new signature in our protocol would then produce the same error you are seeing in versions lower than Xcode 16 beta 3, we will need to under the hood call the both the old and new method. 

I've done some testing in Xcode 16 beta 3 as well as Xcode 15.4 and the change in PR #1360 seems to resolve both issues. If you'd like to pull down that branch and let us know if you're seeing the issue resolved on your end, please feel free.

**Comment by jqsilver** [comment]
I also ran into this today, so I'm happy to see this getting addressed so quickly!

**Comment by FlorianTousch** [comment]
Same here, Bitrise don't build anymore 
![CleanShot 2024-07-10 at 10 46 42@2x](https://github.com/braintree/braintree_ios/assets/5923142/5c53ab6b-7496-4475-aabf-10cb2f0f8496)

**Comment by jaxdesmarais** [comment]
Hey all -

The fix for this issue has been released in [version 6.23.0](https://github.com/braintree/braintree_ios/releases/tag/6.23.0) of the SDK. Thanks again for bringing this to our attention!
